base/cvd/cuttlefish/host/commands/assemble_cvd/flags/bool_flag.cc

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cuttlefish/host/commands/assemble_cvd/flags/bool_flag.h"
+
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include <android-base/strings.h>
+#include <gflags/gflags.h>
+
+#include "cuttlefish/common/libs/utils/flag_parser.h"
+#include "cuttlefish/common/libs/utils/result.h"
+
+namespace cuttlefish {
+
+Result<BoolFlag> BoolFlag::FromGlobalGflagsAndName(
+    const std::string& flag_name) {
+  gflags::CommandLineFlagInfo flag_info =
+      gflags::GetCommandLineFlagInfoOrDie(flag_name.c_str());
```

Comment:
Can the `gflags::CommandLineFlagInfo` instance the argument rather than `flag_name`? That makes the interaction with gflags globals and possible failing assertion (if the caller uses an `...OrDie` function) more obvious.